### PR TITLE
Update search to enable live testing in sovereign clouds for multiple services

### PR DESCRIPTION
**Pull request by v-hongli1**
1.These changes enable search to run live tests against Public, UsGov and China.

2.Update file : SearchSyncTests.java 
  Match the search record order with the verification order, so add sort settings for the test cases `SearchSyncTests.canFilter` and `SearchSyncTests.searchWithScoringProfileBoostsScore`.

3.Update file : test-resources.json
  The number of free search service creations has exceeded the limit, so the `defaultValue` of `searchSku` has been revised to `basic`. 

Pipeline results:
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1078706&view=results

@benbp, @scottaddie, @joshfree, @meeraharidasa, @AlexGhiondea and @alzimmermsft  for notification.

### CODE REVIEW

**Review by benbp — APPROVED** [review]

**Comment by benbp** [comment]
Templates look good but would be good to get a review from a code owner.

**Comment by v-xuto** [comment]
@alzimmermsft  Could you please help review this PR?

**Comment by alzimmermsft** [comment]
> @alzimmermsft Could you please help review this PR?

If the live tests pass this can be merged. Historically, there we return differences that occurred between `free` and `basic` service tiers that may fail.

**Review by alzimmermsft — APPROVED** [review]

**Comment by v-xuto** [comment]
@benbp Do others need to review this PR? If not, maybe we can merge this PR.

File: sdk/search/test-resources.json
Comment (alzimmermsft) [inline sdk/search/test-resources.json:65]: This should be `default` based on the documentation I've read, but it also doesn't apply to the SKU being used. So, not sure if this could cause any issues.

**Comment by alzimmermsft** [comment]
Weekly tests are failing due to an external issue (issue in code), this PR is good to be merged!

**Pull request merged**
